staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/webhook/webhook.go

```diff
@@ -105,7 +106,18 @@ func (w *WebhookTokenAuthenticator) AuthenticateToken(ctx context.Context, token
 	var (
 		result *authenticationv1.TokenReview
 		auds   authenticator.Audiences
+		cancel context.CancelFunc
 	)
+
+	// if the context expires sooner than our hard requestTimeout use it,
+	// otherwise set the request timeout to whatever is defined in requestTimeout
+	if w.requestTimeout > 0 {
+		if oldCtx := ctx; timeBeforeContextDeadline(time.Now().Add(w.requestTimeout), oldCtx) {
```

Comment:
alternatively, we could only set a deadline iff it wasn't set in the `context`

Reply:
I think we can do the following, since contexts are hierarchical:
```
if w.requestTimeout > 0 {
   child, cancel := context.WithTimoeut(ctx,  w.requestTimeout)
   defer cancel()
    
   ctx = child
}
```

if the `child` has a shorter deadline then it will expire first, otherwise if the parent has a shorter deadline then the parent will expire it will cause the child to expire as well even though the child has a larger deadline.

I think that's how it should work, I would also recommend adding a unit test to assert this behavior. :)

Reply:
yes, that would work too :)